src/fondant/core/manifest.py

```diff
@@ -15,60 +15,7 @@
 
 from fondant.core.component_spec import ComponentSpec
 from fondant.core.exceptions import InvalidManifest
-from fondant.core.schema import Field, Type
-
-
-class Subset:
-    """
-    Class representing a Fondant subset.
-
-    Args:
-        specification: The part of the manifest json representing the subset
-        base_path: The base path which the subset location is defined relative to
-    """
-
-    def __init__(self, specification: dict, *, base_path: str) -> None:
-        self._specification = specification
-        self._base_path = base_path
-
-    @property
-    def location(self) -> str:
-        """The absolute location of the subset."""
-        return self._base_path + self._specification["location"]
-
-    @property
-    def fields(self) -> t.Mapping[str, Field]:
-        """The fields of the subset returned as an immutable mapping."""
-        return types.MappingProxyType(
-            {
-                name: Field(name=name, type=Type.from_json(field))
-                for name, field in self._specification["fields"].items()
-            },
-        )
-
-    def add_field(self, name: str, type_: Type, *, overwrite: bool = False) -> None:
-        if not overwrite and name in self._specification["fields"]:
-            msg = f"A field with name {name} already exists"
-            raise ValueError(msg)
-
-        self._specification["fields"][name] = type_.to_json()
-
-    def remove_field(self, name: str) -> None:
-        del self._specification["fields"][name]
-
-    def __repr__(self) -> str:
-        return f"{self.__class__.__name__}({self._specification!r})"
-
-
-class Index(Subset):
-    """Special case of a subset for the index, which has fixed fields."""
-
-    @property
-    def fields(self) -> t.Dict[str, Field]:
-        return {
-            "id": Field(name="id", type=Type("string")),
-            "source": Field(name="source", type=Type("string")),
-        }
+from fondant.core.schema import Field, ManifestField, Type
```

Comment:
If we have a separate `ManifestField`, we probably don't want to import `Field` here to make sure it is not used.

Reply:
I've removed the `ManifestField` again.